### PR TITLE
Glooctl check

### PR DESCRIPTION
**Pull request by rickducott**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1052

### CODE REVIEW

**Comment by rickducott** [comment]
See #1052 for example outputs

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+	cliutils.ApplyOptions(cmd, optionsFunc)
+	return cmd
+}
+
+func checkResources(opts *options.Options) (bool, error) {
+	err := checkConnection()
+	if err != nil {
+		return false, err
+	}
+	ok, err := checkPods(opts)
```
Comment (mitchdraft) [inline projects/gloo/cli/pkg/cmd/check/root.go]: this `ok, err` check pattern suggests an opportunity for an interface with a special check function - can iterate on that as we go

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+		}
+		for _, proxy := range proxies {
+			if proxy.Status.GetState() == core.Status_Rejected {
+				fmt.Printf("Found rejected proxy: %s\n", renderMetadata(proxy.GetMetadata()))
+				fmt.Printf("Reason: %s", proxy.Status.Reason)
+				return false, nil
+			}
+		}
+	}
+	fmt.Printf("OK\n")
```
Comment (mitchdraft) [inline projects/gloo/cli/pkg/cmd/check/root.go]: nit: would be nice to abstract these status reports in a method/interface for consistency - can be done in an iteration

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+	"github.com/solo-io/gloo/projects/gloo/cli/pkg/helpers"
+	v1 "github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
+	"github.com/solo-io/gloo/projects/gloo/pkg/defaults"
+	"github.com/solo-io/go-utils/cliutils"
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients"
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
+	"github.com/spf13/cobra"
+	v13 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (mitchdraft) [inline projects/gloo/cli/pkg/cmd/check/root.go]: probably worth giving these better aliases

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+	return renderNamespaceName(ref.Namespace, ref.Name)
+}
+
+func renderNamespaceName(namespace, name string) string {
+	return fmt.Sprintf("%s %s", namespace, name)
+}
+
+// Checks whether the cluster that the kubeconfig points at is available
+// The timeout for the kubernetes client is set to a low value to notify the user of the failure
+func checkConnection() error {
```
Comment (mitchdraft) [inline projects/gloo/cli/pkg/cmd/check/root.go]: good idea

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+	ok, err = checkProxies(namespaces)
+	if !ok || err != nil {
+		return ok, err
+	}
+
+	return true, nil
+}
+
+func checkPods(opts *options.Options) (bool, error) {
```
Comment (mitchdraft) [inline projects/gloo/cli/pkg/cmd/check/root.go]: might be able to use some of these utils here https://github.com/solo-io/go-utils/tree/master/debugutils
(noting for future work)

**Comment by rickducott** [comment]
```
Step #17 - "regression-tests-cluster-lock":   [91mExpected error:
Step #17 - "regression-tests-cluster-lock":       <*xerrors.wrapError | 0xc00044a080>: {
Step #17 - "regression-tests-cluster-lock":           msg: "deploying testrunner",
Step #17 - "regression-tests-cluster-lock":           err: {
Step #17 - "regression-tests-cluster-lock":               ErrStatus: {
Step #17 - "regression-tests-cluster-lock":                   TypeMeta: {Kind: "", APIVersion: ""},
Step #17 - "regression-tests-cluster-lock":                   ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},
Step #17 - "regression-tests-cluster-lock":                   Status: "Failure",
Step #17 - "regression-tests-cluster-lock":                   Message: "pods \"testrunner\" is forbidden: error looking up service account test-ffd90f10-4ac4-4247-b48b-ccfc45293bf7/default: serviceaccount \"default\" not found",
Step #17 - "regression-tests-cluster-lock":                   Reason: "Forbidden",
Step #17 - "regression-tests-cluster-lock":                   Details: {Name: "testrunner", Group: "", Kind: "pods", UID: "", Causes: nil, RetryAfterSeconds: 0},
Step #17 - "regression-tests-cluster-lock":                   Code: 403,
Step #17 - "regression-tests-cluster-lock":               },
Step #17 - "regression-tests-cluster-lock":           },
Step #17 - "regression-tests-cluster-lock":           frame: {
Step #17 - "regression-tests-cluster-lock":               frames: [0x106b821, 0x106d2c0, 0x12300fc],
Step #17 - "regression-tests-cluster-lock":           },
Step #17 - "regression-tests-cluster-lock":       }
Step #17 - "regression-tests-cluster-lock":       deploying testrunner: pods "testrunner" is forbidden: error looking up service account test-ffd90f10-4ac4-4247-b48b-ccfc45293bf7/default: serviceaccount "default" not found
Step #17 - "regression-tests-cluster-lock":   not to have occurred[0m
```

**Pull request merged**